subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/BuildParamCheckPlugin.kt

```diff
@@ -16,70 +17,106 @@ import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.artifacts.UnknownConfigurationException
 import org.gradle.api.invocation.Gradle
+import org.gradle.kotlin.dsl.create
 import org.gradle.kotlin.dsl.invoke
 import org.gradle.kotlin.dsl.register
 import org.gradle.tooling.BuildException
 
-@Suppress("unused", "RemoveCurlyBracesFromTemplate", "UnstableApiUsage")
+@Suppress("unused")
 open class BuildParamCheckPlugin : Plugin<Project> {
 
     override fun apply(project: Project) {
+        val extension = project.extensions.create<BuildChecksExtension>(extensionName)
+
         printBuildEnvironment(project)
 
         check(project.isRoot()) {
             "Plugin must be applied to the root project but was applied to ${project.path}"
         }
-        val isEnabled = project.getBooleanProperty("avito.build.paramCheck.enabled", false)
-        registerRequiredTasks(project, isEnabled)
+        project.afterEvaluate {
+            val checks = ChecksFilter(project, extension).checks()
+            checks
+                .filterIsInstance<BuildChecksExtension.RequireParameters>()
+                .forEach {
+                    it.validate()
```

Comment:
Looking at the validate method I think we can create ValidateInterface in the future. It will help us to look through extension with type Validate and call `validate()` on each.
Not a blocker